### ISSUE TITLE
MVC AdjustToContents and Width isn't working when converted to a stream

### ISSUE THREAD

**Issue by Martin-Adam** [ClosedXML/ClosedXML#1306]
## Read and complete the full issue template

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature

If you are logging a possible bug or feature request, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**

0.94.2

**What is the current behavior?**

No columns change size based on the set Width or AdjustToContents (besides columns().width as shown in the code) when it is translated to a stream

**What is the expected behavior or new feature?**

It did not adjust to the desired width

**Did this work in previous versions of our tool?  Which versions?**

Unknown

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

Without a code sample, it is unlikely that your issue will get attention. Don't be lazy. Do the effort and assist the developers to reproduce your problem. Code samples should be [minimal complete and verifiable](https://stackoverflow.com/help/mcve). Sample spreadsheets should be attached whenever applicable. Remove sensitive information.

**Code to reproduce problem:**
```c#
public void Main()
{
    // Code standards:
    // - Fully runnable. I should be able to copy and paste this code into a 
    //   console application and run it without having to edit it much.
    // - Declare all your variables (this follows from the previous point)
    // - The code should be a minimal code sample to illustrate issue. The code 
    //   samples on the wiki are good examples of the terseness that I want. Don't
    //   post your full application
    using (var workbook = new XLWorkbook())
    {
        var ws = workbook.Worksheets.Add("Numbers");
        var ws2 = workbook.Worksheets.Add("List");

        #region Numbers Worksheet
        ws.Cell("A1").Value = "Total";
        ws.Cell("B1").Value = 1;
        #endregion

        #region List Worksheet
        #region Header
        ws2.Cell("A1").Value = "Full Name";
        ws2.Cell("B1").Value = "Person Stuff";
        ws2.Cell("C1").Value = "Phone";
        ws2.Cell("D1").Value = "Address";
        #endregion

        #region Data
        ws2.Cell("A2").Value = "That Guy";
        ws2.Cell("B2").Value = "Duffle Bag";
        ws2.Cell("C2").Value = "123-456-7890";
        ws2.Cell("D2").Value = "123 fake street, Where California";
        #endregion

        //Set Table
        var rngData = ws2.Range("A1:D2");
        var excelTable = rngData.CreateTable();
        #endregion


        //Works
        ws.Columns().Width = 20;
        //Doesn't work
        ws.Columns(1, 2).Width = 40;
        //Doesn't work
        ws.Column(3).Width = 20;
        //Doesn't work
        ws2.Columns().AdjustToContents();

        DateTime date = DateTime.Now;
        string path = Environment.GetFolderPath(Environment.SpecialFolder.Desktop);
        workbook.SaveAs(path + "\\TripStatus_" + date.ToString("yyyy_MM_dd") + ".xlsx");
                    
        System.IO.Stream spreadsheetStream = new System.IO.MemoryStream();
        workbook.SaveAs(spreadsheetStream);
        spreadsheetStream.Position = 0;

        //Doing this in MVC not sure how to output this part properly with provided example
        //return new FileStreamResult(spreadsheetStream, "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet") { FileDownloadName = "TripStatus_" + date.ToString("yyyy_MM_dd") + ".xlsx" };
    }
}
```
- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
Stream:
[TripStatus_2019_11_06.xlsx](https://github.com/ClosedXML/ClosedXML/files/3816023/TripStatus_2019_11_06.xlsx)
Save As:
[TripStatus_2019_11_06_SaveAs.xlsx](https://github.com/ClosedXML/ClosedXML/files/3816132/TripStatus_2019_11_06_SaveAs.xlsx)

**Comment by Martin-Adam** [comment]
I am unsure now if stream is the issue.
I had a class that I used to create it, and once I merged over my original code to fit the working solution it didn't work. but when I moved the code into the ActionResult (MVC) it worked perfectly.

**Comment by igitur** [comment]
Do you experience the problem when using streams in a pure console application?

**Comment by Martin-Adam** [comment]
@igitur Thank you for taking your time to get back to me. As discovered below I don't think its a 'stream' issue any longer. Would you like me to close this ticket and open another?

The 'FileStreamResult' seems to be in the MVC library. However I tested the other thing I discovered.

As for me thinking its no long a stream issue I seem to be correct here is the code I just tried

Having it go to another class with the workbook seems to cause some kind of issue

```c#
using ClosedXML.Excel;
using System;
using System.IO;

namespace ConsoleApp1
{
    class Program
    {
        static void Main(string[] args)
        {
            DateTime date = DateTime.Now;
            Excel e = new Excel();
            System.IO.Stream spreadsheetStream = new System.IO.MemoryStream();
            XLWorkbook workbook = e.wb();   //This gets/returns the workbook
            string path = Path.GetTempPath();
            string path2 = Environment.GetFolderPath(Environment.SpecialFolder.Desktop);
            string fileName = "\\TripStatus_" + date.ToString("yyyy_MM_dd") + ".xlsx";
            workbook.SaveAs(path2 + fileName);
        }
    }
}
```

```c#
using ClosedXML.Excel;

namespace ConsoleApp1
{
    class Excel
    {
        public XLWorkbook wb()
        {
            using (var workbook = new XLWorkbook())
            {
                var ws = workbook.Worksheets.Add("Numbers");
                var ws2 = workbook.Worksheets.Add("List");

                #region Numbers Worksheet
                ws.Cell("A1").Value = "Total";
                ws.Cell("B1").Value = 1;
                #endregion

                #region List Worksheet
                #region Header
                ws2.Cell("A1").Value = "Full Name";
                ws2.Cell("B1").Value = "Person Stuff";
                ws2.Cell("C1").Value = "Phone";
                ws2.Cell("D1").Value = "Address";
                #endregion

                #region Data
                ws2.Cell("A2").Value = "That Guy";
                ws2.Cell("B2").Value = "Duffle Bag";
                ws2.Cell("C2").Value = "123-456-7890";
                ws2.Cell("D2").Value = "123 fake street, Where California";
                #endregion

                //Set Table
                var rngData = ws2.Range("A1:D2");
                var excelTable = rngData.CreateTable();
                #endregion


                //Works
                ws.Columns().Width = 20;
                //Doesn't work
                ws.Columns(1, 2).Width = 40;
                //Doesn't work
                ws.Column(3).Width = 20;
                //Doesn't work
                ws2.Columns().AdjustToContents();
                return workbook;
            }
        }
    }
}
```

[TripStatus_2019_11_07.xlsx](https://github.com/ClosedXML/ClosedXML/files/3820459/TripStatus_2019_11_07.xlsx)

**Comment by Pankraty** [comment]
>`XLWorkbook workbook = e.wb();   //This gets/returns the workbook`

This gets/returns the _disposed_ workbook as it is defined in a `using` scope inside the method.

Replace `using (var workbook = new XLWorkbook())` with `var workbook = new XLWorkbook();` and this will solve your issue.

@igitur I guess we have to throw `ObjectDisposedException` on certain operations on a workbook. Agree?

**Comment by igitur** [comment]
@Pankraty Agree, but I'm a novice with regards to that exception. How would that be implemented?

**Comment by Pankraty** [comment]
Nothing special, something like this

```
public void Save()
{
  AssertNotDisposed();

  //the rest of the code
}

private void AssertNotDisposed()
{
  if (_disposed)
    throw new ObjectDisposedException();
}

```

In all publicly accessible methods. It is a lot of boilerplate code but it would make apparent the reason for incorrect behavior.

**Comment by igitur** [comment]
Oh. I assumed that if an object is disposed, its methods aren't accessible. Never even tried.

**Comment by igitur** [comment]
Fixed in #1499
